x-pack/plugins/alerting/server/lib/alert_instance.ts

```diff
@@ -45,6 +45,9 @@ export class AlertInstance {
   }
 
   fire(actionGroup: string, context: Context = {}) {
+    if (this.shouldFire()) {
```

Comment:
Left a comment wondering if we need a guard to prevent infinite retries when the condition occurs, or maybe log instead of throwing an error.

Also, looking at `AlertInstance::shouldFire()`, wondering if the implementation should just be the following, removing the need for the internalShouldFire instance property:

```js
  shouldFire() {
    return this. fireOptions != null;
  }
```

Reply:
Yeah, that makes sense, I will do the change